src/util.rs

```diff
@@ -62,3 +66,36 @@ impl LazyBool {
         self.0.unsync_init(|| init() as usize) != 0
     }
 }
+
+/// Polyfill for `maybe_uninit_slice` feature's
+/// `MaybeUninit::slice_assume_init_mut`. Every element of `slice` must have
+/// been initialized.
+#[inline(always)]
+pub unsafe fn slice_assume_init_mut<T>(slice: &mut [MaybeUninit<T>]) -> &mut [T] {
+    // SAFETY: `MaybeUninit<T>` is guaranteed to be layout-compatible with `T`.
+    &mut *(slice as *mut [MaybeUninit<T>] as *mut [T])
```

Comment:
It looks like this function is instantiated only with `T` being equal to `u8`, no? Is it worth to keep it generic? Same for other functions in this module.

Reply:
The functions that are being polyfilled are generic so I left these as generic.

Reply:
For this reason, I think its fine to leave these as generic.